homeassistant/components/crownstone/config_flow.py

```diff
@@ -0,0 +1,296 @@
+                self.updated_entry_data[CONF_USB_PATH] = None
+                self.updated_entry_data[CONF_USB_SPHERE] = None
+                user_input.pop(CONF_USB_SPHERE_OPTION)
+            elif (
+                CONF_USB_SPHERE_OPTION in user_input
+                and spheres[user_input[CONF_USB_SPHERE_OPTION]] != usb_sphere
+            ):
+                sphere_id = spheres[user_input[CONF_USB_SPHERE_OPTION]]
+                user_input[CONF_USB_SPHERE_OPTION] = sphere_id
+                self.updated_entry_data[CONF_USB_SPHERE] = sphere_id
+
+            self.entry.data = MappingProxyType(self.updated_entry_data)
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(step_id="init", data_schema=options_schema)
+
+    async def async_step_usb_config_option(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Set up a Crownstone USB dongle."""
+        list_of_ports = await self.hass.async_add_executor_job(
+            serial.tools.list_ports.comports
+        )
+        ports_as_string = list_ports_as_str(list_of_ports, False)
+
+        if user_input is not None:
+            selection = user_input[CONF_USB_PATH]
+
+            if selection == MANUAL_PATH:
+                return await self.async_step_usb_manual_config_option()
+            if selection != REFRESH_LIST:
+                selected_port: ListPortInfo = list_of_ports[
+                    ports_as_string.index(selection)
+                ]
+                usb_path = await self.hass.async_add_executor_job(
+                    usb.get_serial_by_id, selected_port.device
+                )
+                self.updated_entry_data[CONF_USB_PATH] = usb_path
+                return await self.async_step_usb_sphere_config_option()
+
+        return self.async_show_form(
+            step_id="usb_config_option",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_USB_PATH): vol.In(ports_as_string)}
+            ),
+        )
+
+    async def async_step_usb_manual_config_option(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manually enter Crownstone USB dongle path."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="usb_manual_config_option",
+                data_schema=vol.Schema({vol.Required(CONF_USB_MANUAL_PATH): str}),
+            )
+
+        self.updated_entry_data[CONF_USB_PATH] = user_input[CONF_USB_MANUAL_PATH]
+        return await self.async_step_usb_sphere_config_option()
+
+    async def async_step_usb_sphere_config_option(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Select a Crownstone sphere that the USB operates in."""
+        # no need to select if there's only 1 option
+        if len(self.spheres) == 1:
+            user_input = {CONF_USB_SPHERE: next(iter(self.spheres.keys()))}
```

Comment:
Can you avoid replacing `user_input` here? When doing a security review it makes it harder to tell which data is coming from the user and which data is trusted.